lib/services/libvirtd.pm

```diff
@@ -0,0 +1,132 @@
+# SUSE's openQA tests
+#
+# Copyright © 2021 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Package for libvirtd service check tests for migration
+#
+# Maintainer: Wegao <wegao@suse.com>
+
+package services::libvirtd;
+use base 'opensusebasetest';
+use testapi;
+use utils;
+use strict;
+use warnings;
+
+#default guest vm params
+our %guest_params = (
+    'name'        => 'nested_vm_test',
+    'ram' => '',
+    'memory' => '512',
+    'vcpus'        => '',
+    'os-type'  => '',
+    'disk'  => 'none',
+    'network'  => '',
+    'graphics'  => 'vnc',
+    'boot'  => 'cdrom',
+);
+
+sub remove_repo {
+    my $repo_max_num = script_output("zypper lr | tail -1 | cut -d' ' -f1");
+    zypper_call("lr -u");
+    for (my $i = 1; $i < $repo_max_num + 1; $i = $i +1) {
+        zypper_call("rr 1");
+    }
+    script_run("zypper lr -u");
+}
+sub install_service {
+    if (get_var("DISTRI") eq "opensuse") {
+        remove_repo();
+        my $version = script_output("grep VERSION= /etc/os-release | cut -d'=' -f2 | cut -d' ' -f1 | sed 's/\"//g'");
+        zypper_call("ar http://download.opensuse.org/distribution/leap/$version/repo/oss/ main");
+        zypper_call("ar http://download.opensuse.org/update/leap/$version/oss/ mainupdate");
```

Comment:
I guess this is meant to be a hardcoded test for leap right?, if so would you mind adding this for TW?

Reply:
I only find upgrade case from leap->TW in openqa.opensuse.org
I can not find any case for TW to TW, do we have this kind of check?

Reply:
@foursixnine ?

Reply:
Was just a question, I don't think we have TW to TW for now we need to introduce within qe-core [poo#16618](https://progress.opensuse.org/issues/16618) for that.

Reply:
Notify me or create ticket for me once  TW to TW is ready, then we can integrate it.